front/src/common/useSearchUser.ts

```diff
@@ -0,0 +1,64 @@
+import { useState, useCallback, useEffect } from 'react';
+
+import { DEFAULT_GRANT } from 'modules/infra/consts';
+import { useDebounce } from 'utils/helpers';
+
+import type { SearchResultItemUser } from './api/generatedEditoastApi';
+import type { SubjectItemWithGrant } from './api/mock/mockEditoastApi';
+import { osrdEditoastApi } from './api/osrdEditoastApi';
+
+type SearchUser = {
+  debounceDelay?: number;
+  initialSearchTerm?: string;
+};
+
+export default function useSearchUser({
```

Comment:
It would be better to name this hook searchUsers (plural), what do you think ?
```suggestion
export default function useSearchUsers({
```